src/Data/Text/AhoCorasick/Automaton.hs

```diff
@@ -4,12 +4,6 @@
 -- Licensed under the 3-clause BSD license, see the LICENSE file in the
 -- repository root.
 
--- Compile this module with LLVM, rather than with the default code generator.
--- LLVM produces about 20% faster code.
--- We pass -fignore-asserts to improve performance: we ran this code with
--- asserts enabled in production for two months, and in this time, the asserts
--- have not been violated.
-{-# OPTIONS_GHC -fllvm -O2 -optlo=-O3 -optlo=-tailcallelim -fignore-asserts #-}
```

Comment:
It looks like we lose the `-O2` now. Is it safe to remove? I recall something about Stack building with `-O2` by default but I don't remember the details.

Reply:
That flag should also be redundant. [Cabal uses `-O2` by default](https://www.haskell.org/cabal/release/cabal-2.4.1.0/doc/users-guide/installing-packages.html#cmdoption-setup-configure-enable-optimization) and you explicitly need to turn optimizations off if you don't want them.

I'm not sure how those flags propagate to any dependencies though. We can add `-O2` to the global ghc options in the cabal file to be sure.